spec/functional/resource/dpkg_package_spec.rb

```diff
@@ -45,11 +45,19 @@ def set_dpkg_package_name(name)
     dpkg_package.package_name name
   end
 
-  def should_be_purged_or_removed(package, action)
+  def should_be_purged_or_removed(package, action=nil)
+    status = shell_out("dpkg -s #{package}")
+    output = status.stdout + status.stderr
     if action == :purge
-      shell_out!("dpkg -s #{package}", returns: [1])
+      not_installed = !(output =~ /deinstall/) &&
+        (!!(output =~ /not installed/) || !!(output =~ /not-installed/))
+      expect(not_installed).to be(true)
+    elsif action == :remove
+      expect(output).to match(/deinstall ok config-files/)
+    elsif action == nil
+      expect(output).to match(/deinstall|no info|not-installed/)
```

Comment:
If you made this `match(/no info|not-installed/not installed/)` then I think it would work and you could use this for both the :purge case and this nil case.